Sources/SwiftDocCUtilities/Action/Actions/Convert/ConvertAction.swift

```diff
@@ -462,14 +462,18 @@ public struct ConvertAction: Action, RecreatingContext {
         }
 
         var didEncounterError = analysisProblems.containsErrors || conversionProblems.containsErrors
-        if try context.renderRootModules.isEmpty {
+        if (try context.renderRootModules + context.rootTechnologies).isEmpty {
```

Comment:
I was reading the code behind `renderRootModules` and `rootTechnologies`, and looks like they return the nodes in the topic graph that have kind=module or kind=technology. But there's also a static DocumentationNode.Kind called `.root`:

```
/// Root-level documentation.
public static let root = DocumentationNode.Kind(name: "Documentation", id: "org.swift.docc.kind.root", isSymbol: false)
```

Does DocC use this node type? Do we need to consider those root nodes here?

Reply:
> Does DocC use this node type? Do we need to consider those root nodes here?

I didn't find anywhere that this kind was assigned to a node, so maybe it was implemented there originally but never used (?). Maybe @d-ronnqvist can better explain if that kind is meant to be used anywhere.

I'm following the same logic used in the previewPaths() method, so I don't think we're missing any other root type.

Reply:
I don't _think_ that we use `root` for anything. We only have 3 types of top-level pages:
- modules (frameworks/libraries/apps)
- technology root pages
- tutorial table of content pages

Modules use `.module` and IIRC tutorial table of content pages use `.technology` (The implementation calls them "technologies" and it's a bit confusing that it's different from the user-facing terminology). 

AFAICT technology root pages also use `.module` but I don't know if that's a bug or if it's intended behavior.